### PR TITLE
Add support for Weatherbit in the Weather Module

### PR DESCRIPTION
**Pull request by AndyPoms**
This is my first ever pull request, please let me know if I messed anything up.

Adds support for Weatherbit.io
Based on existing classes for Dark Sky and Weather.gov

***API Key required***

Config Options:

Option | Description
---|---
`apiBase` | The Weather base URL.   Possible value: `https://api.weatherbit.io/v2.0`  This value is **REQUIRED**
`weatherEndpoint` | The Weatherbit API endPoint.  Possible values: `/current`, `/forecast/daily`  This value is **REQUIRED**
`apiKey` | The Weatherbit API key which can be obtained by creating an WeatherBit account at https://www.weatherbit.io  This value is **REQUIRED**
`lat` | The geo coordinate latitude.  This value is **REQUIRED**
`lon` | The geo coordinate longitude.  This value is **REQUIRED**

### CODE REVIEW

File: modules/default/weather/providers/weatherbit.js [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+	// Create a URL from the config and base URL.
+	getUrl() {
+		const units = this.units[this.config.units] || "auto";
+		return `${this.config.apiBase}${this.config.weatherEndpoint}?lat=${this.config.lat}&lon=${this.config.lon}&units=${units}&key=${this.config.apiKey}`;
+	},
+
+	// Implement WeatherDay generator.
+	generateWeatherDayFromCurrentWeather(currentWeatherData) {
+		//Calculate TZ Offset and invert to convert Sunrise/Sunset times to Local
+		var d = new Date();
```
Comment (rejas) [inline modules/default/weather/providers/weatherbit.js:71]: can be a const ?

File: modules/default/weather/providers/weatherbit.js [hunk 2]
```diff
@@ -0,0 +1,181 @@
...
+		var tzOffset = d.getTimezoneOffset();
```
Comment (rejas) [inline modules/default/weather/providers/weatherbit.js:72]: should be a let

**Comment by rejas** [comment]
As the bot says: Please include an entry for your PR in the Changelog. 
Otherwise I found only two small nitbits looking at the code, dont have the time unfortanetly to run it and see if it works :-)

**Comment by rejas** [comment]
Also, you should run "npm run lint:prettier" to make sure your code adheres to the styleguides (and passes the travis tests)

**Comment by AndyPoms** [comment]
Like I said this is my first ever pull request...  How do I include a CHANGELOG.md file?  What should it include? 

Do I just take the one from my fork, add a line that says "Add support for Weatherbit" & attach?  Do I create a new file named CHANGELOG.md with only a line that say "Adds support for Weatherbit"?

How do I run `npm run lint:prettier`?  Is it on the website, the Desktop app, or GitBash, GitCMD, or GitGUI?

@rejas I'll review your const/let comments when I get a chance (likely towards the end of the week).

**Comment by rejas** [comment]
> Like I said this is my first ever pull request... How do I include a CHANGELOG.md file? What should it include?
> 
> Do I just take the one from my fork, add a line that says "Add support for Weatherbit" & attach? Do I create a new file named CHANGELOG.md with only a line that say "Adds support for Weatherbit"?

You should just add a new line to the already existing CHANGELOG.md file under the "Added" headline.

> How do I run `npm run lint:prettier`? Is it on the website, the Desktop app, or GitBash, GitCMD, or GitGUI? Or is it on the machine I'm hosting my MM with?
You run that command on the command line on the machine you are developing and submitting your PR. Just go into the main directory of the MagicMirror repository.

> @rejas I'll review your const/let comments when I get a chance (likely towards the end of the week).
:+1:

**Comment by sdetweil** [comment]
>>How do I run npm run lint:prettier

it's a commandline
all git and npm commands are from the MagicMirror folder

**Comment by AndyPoms** [comment]
- Updated to const & let per rejas suggestions.
- Added CHANGELOG.md (I think I may have screwed this one up based on the Conflict listed above - advice?)
- Ran `npm run lint:prettier`

Let me know if there is anything else

**Comment by rejas** [comment]
Just fix that one merge-conflict and the PR seems good to go!

**Review by rejas — APPROVED** [review]

**Comment by AndyPoms** [comment]
> 
> 
> Just fix that one merge-conflict and the PR seems good to go!

I think I did it correctly & kept all changes (just removed the 3 conflict markers so that all info remained.

**Comment by rejas** [comment]
Please run "npm run lint:prettier" to fix the style issues in the CHANGELOG. After those changes are pushed, this PR should be fine.

**Comment by AndyPoms** [comment]
> 
> 
> Please run "npm run lint:prettier" to fix the style issues in the CHANGELOG. After those changes are pushed, this PR should be fine.

Done....  Weird because my original CHANGELOG was OK, it was only after I resolved a conflict (by keeping both sets of changes) that this came up....

**Pull request merged**